polaris/polaris/management/commands/process_pending_deposits.py

```diff
@@ -35,151 +35,158 @@
 )
-from polaris.models import Transaction
+from typeguard import check_protocol
+
+from polaris.exceptions import (
+    TransactionSubmissionPending,
+    TransactionSubmissionBlocked,
+    TransactionSubmissionFailed,
+)
+
+from polaris.models import Transaction, PolarisHeartbeat
 from polaris.utils import getLogger
 
+
 logger = getLogger(__name__)
 
-TERMINATE = False
-"""
-SIGINT and SIGTERM signals to this process set TERMINATE to True,
-and once all pending tasks complete, the process exits.
-Only relevant if the --loop option is specified.
-"""
+SUBMIT_TRX_QUEUE = "SUBMIT_TRX_QUEUE"
+CHECK_ACC_QUEUE = "CHECK_ACC_QUEUE"
 
+DEFAULT_HEARTBEAT = 5
 DEFAULT_INTERVAL = 10
-"""
-The default amount of time to sleep before querying for transactions again
-Only used if the --loop option is specified.
-"""
 
+PROCESS_PENDING_DEPOSITS_LOCK_KEY = "PROCESS_PENDING_DEPOSITS_LOCK"
 
-class Command(BaseCommand):
-    """
-    The process_pending_deposits command handler.
-    """
+# TODO: undo change made to docker-compose.yaml
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        signal.signal(signal.SIGINT, self.exit_gracefully)
-        signal.signal(signal.SIGTERM, self.exit_gracefully)
 
-    @staticmethod
-    def exit_gracefully(*_):  # pragma: no cover
-        logger.info("Exiting process_pending_deposits...")
-        module = sys.modules[__name__]
-        module.TERMINATE = True
+# TODO: move this
+class QueueAdapter:
+    def populate_queues(self):
+        """
+        Initialize in-memory queues from database, this is optional if you're using an actual 
+        message queue or storing your "queues" in a database
+        """
+        raise NotImplementedError
 
-    @staticmethod
-    def sleep(seconds):  # pragma: no cover
-        for _ in range(seconds):
-            if TERMINATE:
-                break
-            time.sleep(1)
+    def queue_transaction(
+        self, source_task_name: str, queue_name: str, transaction: Transaction
+    ):
+        """
+        Put the given transaction into a queue
+        @param: source_task_name - the task that queued this transaction
+        @param: queue_name - name of the queue to put the Transaction in
+        @param: transaction - the Transaction to put in the queue
+        """
+        raise NotImplementedError
 
-    def add_arguments(self, parser):  # pragma: no cover
-        parser.add_argument(
-            "--loop",
-            action="store_true",
-            help="Continually restart command after a specified number of seconds.",
+    async def get_transaction(self, source_task_name: str, queue_name) -> Transaction:
+        """
+        Consume a transaction from a queue
+        @param: source_task_name - the task that is requesting a Transaction
+        @param: queue_name - name of the queue to consume the Transaction from
+        """
+        raise NotImplementedError
+
+
+class PolarisQueueAdapter(QueueAdapter):
```

Comment:
I see, but thats not possible right now since we're not allowing developers to register queue integrations. Can we consolidate this functionality into a single class that Polaris uses and then file an issue kicking off the discussion of allowing different queue implementations?

Reply:
sure